main.tf

```
@@ -200,9 +232,63 @@ resource "aws_iam_role_policy_attachment" "lambda_datadog_push" {
 resource "aws_cloudwatch_log_group" "log_group" {
   name              = "/aws/lambda/${aws_lambda_function.logs_to_datadog.function_name}"
   retention_in_days = var.retention
+  kms_key_id        = aws_kms_alias.datadog.arn
   tags              = local.tags
 }
 
+data "aws_iam_policy_document" "cloudwatch_logs_kms_policy" {
+
+  statement {
+    sid = "AllowCloudWatchtoUseKMSKey"
+    #tfsec:ignore:aws-iam-no-policy-wildcards
+    actions = [
+      "kms:Encrypt*",
+      "kms:Decrypt*",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:Describe*"
+    ]
+    #tfsec:ignore:aws-iam-no-policy-wildcards
+    resources = [
+      "*"
+    ]
```

Technically this would never have a `,` as it already includes all (`*`) so you should just make it a 1 liner?

```suggestion
    resources = ["*"]
```